### PR TITLE
[DOXIA-606] fix javadoc issues with JDK 8 when generating documentation

### PR DESCRIPTION
**Pull request by michaelboyles**
https://issues.apache.org/jira/browse/DOXIA-606

Boy does this thing take a while to build!

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  24:58 min
[INFO] Finished at: 2020-09-14T18:12:47+01:00
[INFO] ------------------------------------------------------------------------
```

### CODE REVIEW

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java [hunk 1]
```diff
@@ -340,6 +340,7 @@ public void testHtmlContent()
      * @param file the file to parse with {@link #parser}.
      * @return a sink to test parsing events.
      * @throws ParseException if the document parsing failed.
+     * @throws IOException if an I/O error occurs while closing test reader.
```
Comment (elharo) [inline doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java:343]: no period at end of doc comment unless it's a complete sentence; per Oracle style guide

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java [hunk 2]
```diff
@@ -361,7 +362,7 @@ protected String parseFileToHtml( String file ) throws ParseException, IOExcepti
         }
     }
 
-    /** @throws Exception  */
+    /** @throws Exception if the document parsing failed. */
```
Comment (elharo) [inline doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java:365]: no  period, or just delete this one. We don't need doc comments for tests

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java [hunk 3]
```diff
@@ -388,7 +389,7 @@ public void testTocMacro()
     /**
      * TOC macro fails with EmptyStackException when title 2 followed by title 4 then title 2
      * 
-     * @throws Exception
+     * @throws Exception if the document parsing failed.
```
Comment (elharo) [inline doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java:392]: ditto

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java [hunk 4]
```diff
@@ -420,7 +421,7 @@ public void testTocMacroDoxia559()
     /**
      * test fix for https://github.com/vsch/flexmark-java/issues/384
      *
-     * @throws Exception
+     * @throws Exception if the document parsing failed.
```
Comment (elharo) [inline doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java:424]: ditto

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java (hunk 1 above)
Comment (michaelboyles) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java:343]: I copied the existing precedent. Would you rather I change the other ones which were not affecting the build or make them inconsistent with one another?

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java (hunk 4 above)
Comment (michaelboyles) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java]: Moved description to regular comment

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java (hunk 3 above)
Comment (michaelboyles) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java]: Moved description to regular comment

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java (hunk 2 above)
Comment (michaelboyles) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java]: Deleted

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java (hunk 1 above)
Comment (elharo) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java:343]: Up to you, but please don't introduce new violations

Comment (michaelboyles) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownParserTest.java:343]: Ok, will fix the others

**Review by elharo — APPROVED** [review]

**Review by elharo — COMMENTED** [review]
running through jenkins: https://ci-builds.apache.org/job/Maven/job/maven-box/job/maven-doxia/job/38/

**Review by slachiewicz — COMMENTED** [review]
Thx, merged.
We still have some issiues with package.html file and generated modello classes.

**Pull request closed without merging**